src/test/java/uk/ac/ebi/ampt2d/persistence/repository/FileMetadataRepositoryTest.java

```diff
@@ -0,0 +1,120 @@
+/*
+ * Copyright 2017 EMBL - European Bioinformatics Institute
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package uk.ac.ebi.ampt2d.persistence.repository;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.test.context.junit4.SpringRunner;
+import uk.ac.ebi.ampt2d.FileType;
+import uk.ac.ebi.ampt2d.persistence.entities.FileMetadata;
+import uk.ac.ebi.ampt2d.persistence.entities.SourceFilePath;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+import static junit.framework.TestCase.assertNotNull;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNull;
+
+/**
+ * Testing the basic CRUD operations
+ */
+@RunWith(SpringRunner.class)
+@SpringBootTest
+public class FileMetadataRepositoryTest {
+    @Autowired
+    private FileRepository fileRepository;
+
+    @Before
+    public void setUp() throws Exception {
+        fileRepository.deleteAll();
```

Comment:
The .`init` is creating the root dir where the files will be stored during the upload. The .`deleteAll` is removing the dir and the content to make sure it is starting from fresh. In production we will need to define the path in `StorageProperties` class and remove the init Bean